grade/pipeline/partialcredit.py

```diff
@@ -29,6 +29,9 @@ class PartialCredit:
     def __init__(self, pipelines: Iterator[Pipeline], value: Union[int, List[int]]):
         self.pipelines = list(pipelines)
         self.max_score = value
+        if type(value) is list:
+            self.max_score = (sum(value[:len(self.pipelines) % len(value)])
+                              + sum(value) * len(self.pipelines) // len(value))
```

Comment:
I'm curious why this wouldn't just be a sum of the values given? Isn't that what we award when a student passes all pipelines?

Reply:
The way partialcredit is written right now, you can give a list of values that will be repeated in order until all of the pipelines have been run (which seems to be why it is put into a deque), so this code accounts for that, as well as accounting for a number of pipelines that isn't a multiple of the number of values.

Reply:
Fair point! Do we think that is good behavior? Should we restrict usage to only lists of equal length to their pipelines? Do you foresee using the rotating feature of the values?

Reply:
The main case I'm currently using does use that feature. I have a list of points values for each type of test, and I need to run all of those tests 3 times with slightly different inputs, so having a 5 element list instead of a 15 element list with repeats seems better.

Reply:
great! that perfectly justifies leaving this in!